rest-fights/src/main/java/io/quarkus/sample/superheroes/fight/client/WebAuthnResponse.java

```diff
@@ -0,0 +1,77 @@
+package io.quarkus.sample.superheroes.fight.client;
+
+import jakarta.ws.rs.BeanParam;
+import jakarta.ws.rs.FormParam;
+
+import io.vertx.core.json.JsonObject;
+
+import org.jboss.resteasy.reactive.RestForm;
+
+public abstract class WebAuthnResponse {
+
+  /**
+   * Corresponds to the JSON {@code id} field
+   */
+  @FormParam("webAuthnId")
+  public String webAuthnId;
+
+  /**
+   * Corresponds to the JSON {@code rawId} field
+   */
+  @FormParam("webAuthnRawId")
+  public String webAuthnRawId;
+
+  /**
+   * Corresponds to the JSON {@code response.clientDataJSON} field
+   */
+  @FormParam("webAuthnResponseClientDataJSON")
+  public String webAuthnResponseClientDataJSON;
+
+  /**
+   * Corresponds to the JSON {@code type} field
+   */
+  @FormParam("webAuthnType")
+  public String webAuthnType;
+
+  /**
+   * Turns this form-based structure into its corresponding JSON structure
+   */
+  public JsonObject toJsonObject() {
```

Comment:
Why using generic `JsonObject` rather than creating an object that represents the proper structure?

Reply:
https://github.com/quarkusio/quarkus/blob/main/extensions/security-webauthn/runtime/src/main/java/io/quarkus/security/webauthn/WebAuthnResponse.java#L43 Used the same one available here. Since `WebAuthnResponse` is only available if you have the `WebAuthn` extension and I didn't want to add the extension to rest-fights, I copied it from there